ServerCore/Pages/Teams/AllSubmissions.cshtml.cs

```diff
@@ -29,31 +28,40 @@ public AllSubmissionsModel(PuzzleServerContext serverContext, UserManager<Identi
 
         public SortOrder? Sort { get; set; }
 
+        public bool SolutionsOnly { get; set; }
+
         private const SortOrder DefaultSort = SortOrder.GroupAscending;
 
         public async Task OnGetAsync(SortOrder? sort, int teamId)
         {
             TeamID = teamId;
 
             IQueryable<PuzzleStatePerTeam> puzzleStates = _context.PuzzleStatePerTeam
-                .Where((state) => state.TeamID == teamId && state.SolvedTime != null && state.Puzzle.IsPuzzle);
+                .Where((state) => state.TeamID == teamId && state.Puzzle.IsPuzzle);
+            
+            if(SolutionsOnly)
+            {
+                puzzleStates = puzzleStates.Where((state) => state.SolvedTime != null);
+            }
 
             IQueryable<Submission> submissions = _context.Submissions
                 .Where((s) => s.TeamID == teamId);
 
             IQueryable<SubmissionView> finalSubmissions =
                 from state in puzzleStates
-                join submission in submissions on state.PuzzleID equals submission.PuzzleID into joinedStateSubmission
-                from joinedSubmission in joinedStateSubmission.DefaultIfEmpty()
+                join submission in submissions on state.PuzzleID equals submission.PuzzleID //into joinedStateSubmission

```

Comment:
Is this intentional and if so why? Honestly, I find joins impossible to read, so my question is very very very basic here; I have no idea what is being added or lost here.

Reply:
OK thanks for the explanation!